src/main/java/org/folio/inventory/resources/MoveApi.java

```diff
@@ -0,0 +1,134 @@
+package org.folio.inventory.resources;
+
+import static java.util.stream.Collectors.toList;
+import static org.folio.inventory.support.JsonArrayHelper.toListOfStrings;
+import static org.folio.inventory.support.http.server.JsonResponse.unprocessableEntity;
+import static org.folio.inventory.validation.ItemsMoveValidator.itemsMoveHasRequiredFields;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+import org.apache.commons.collections15.ListUtils;
+import org.folio.inventory.common.WebContext;
+import org.folio.inventory.domain.items.Item;
+import org.folio.inventory.domain.items.ItemCollection;
+import org.folio.inventory.storage.Storage;
+import org.folio.inventory.storage.external.CollectionResourceClient;
+import org.folio.inventory.storage.external.CqlQuery;
+import org.folio.inventory.storage.external.MultipleRecordsFetchClient;
+import org.folio.inventory.support.ItemUtil;
+import org.folio.inventory.support.http.client.OkapiHttpClient;
+import org.folio.inventory.support.http.server.JsonResponse;
+import org.folio.inventory.support.http.server.ServerErrorResponse;
+import org.folio.inventory.support.http.server.ValidationError;
+
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+
+public class MoveApi extends AbstractInventoryResource {
+
+  public static final String TO_HOLDINGS_RECORD_ID = "toHoldingsRecordId";
+  public static final String ITEM_IDS = "itemIds";
+
+  public MoveApi(final Storage storage, final HttpClient client) {
+    super(storage, client);
+  }
+
+  @Override
+  public void register(Router router) {
+    router.post("/inventory/items/move")
+      .handler(this::moveItems);
+  }
+
+  private void moveItems(RoutingContext routingContext) {
+
+    WebContext context = new WebContext(routingContext);
+    JsonObject itemsMoveJsonRequest = routingContext.getBodyAsJson();
+
+    Optional<ValidationError> validationError = itemsMoveHasRequiredFields(itemsMoveJsonRequest);
+    if (validationError.isPresent()) {
+      unprocessableEntity(routingContext.response(), validationError.get());
+    } else {
+      String toHoldingsRecordId = itemsMoveJsonRequest.getString(TO_HOLDINGS_RECORD_ID);
+      List<String> itemIdsToUpdate = toListOfStrings(itemsMoveJsonRequest.getJsonArray(ITEM_IDS));
+      try {
+
+        OkapiHttpClient okapiClient = createHttpClient(routingContext, context);
+        CollectionResourceClient itemsStorageClient = createItemsStorageClient(okapiClient, context);
```

Comment:
Take a look at `org.folio.inventory.storage.external.Clients`, you can add a `getItemsStorageClient` method in that class. And then extend this `moveItems` method to `moveItems(RoutingContex, WebContext, Clients)`.

You also need to change the `register` method to:

```
@Override
public void register(Router router) {
router.post("/inventory/items/move")
    .handler(handle(this::moveItems));
}
```

Reply:
@bohdan-suprun are you fine to save current approach to maintain uniformity with other API's implementation? I would like to skip this improvement because MoveApi was implemented based on ItemApi and other resources.

Reply:
sure, no prob. I'm resolving the conversation.